### PR TITLE
deepl-connector

### PR DESCRIPTION
**Pull request by ivy-rew**
initial state:
- [x] translates raw-text
- [x] translates full-documents up to 500K characters (by keeping the formatting) :heart: 
- [x] tested
- [x] very simple demo

### CODE REVIEW

File: deepl-connector/pom.xml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  <version>10.0.0-SNAPSHOT</version>
+  <packaging>iar</packaging>
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>com.axonivy.ivy.ci</groupId>
+        <artifactId>project-build-plugin</artifactId>
+        <version>10.0.1</version>
+        <configuration>
+            <ivyVersion>10.0.4</ivyVersion>
```
Comment (alexsuter) [inline deepl-connector/pom.xml:16]: we should have a property <useLatestMinor>true</useLatestMinor>

I think we would use that for all market artifacts. And we also would need this for our own build infrastructure, that we no longer need to define maven ranges.
Maybe a good task for tgr?

**Review by alexsuter — APPROVED** [review]
Nice connector!

No in deep-review. groupIds and artifacts looks good.

File: deepl-connector-product/pom.xml [hunk 2]
```diff
@@ -26,6 +30,29 @@
...
+            <phase>generate-sources</phase>
+            <configuration>
+              <skip>${skip-readme}</skip>
+              <tasks>
+                <copy todir="${project.build.directory}">
+                  <fileset dir="../../" includes="doc/**" />
+                </copy>
+                <copy file="README.md" tofile="${project.build.directory}/README.md" />
+                <loadfile property="variables.yaml" srcFile="${variables.yaml.file}" encoding="UTF-8" />
+                <replace file="${project.build.directory}/README.md" token="@variables.yaml@" value="${variables.yaml}" />
```
Comment (ivy-rew) [inline deepl-connector-product/pom.xml:47]: I think we should add this pattern replacement to the product-template: 
- almost every product needs configuration and therefore explain their variables ... now we always duplicate the variables for that purpose into the readme, which is prone to errors.
Do you agree @alexsuter

File: deepl-connector-product/README.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+## Setup
+
+Get a free developer account from DeepL.com
+Copy the API-Key of your account into
+
+`config/variables.yaml` under
+`variables.deepl-connector.authKey`
+
+```
+@variables.yaml@
```
Comment (ivy-rew) [inline deepl-connector-product/README.md:18]: here's how the feature looks for the user: damn simple :)

File: deepl-connector/pom.xml (hunk 1 above)
Comment (ivy-rew) [reply deepl-connector/pom.xml:16]: yes, good idea, if we can easily get that information from the website, sure.
I create a story for TGR

Comment (ivy-rew) [reply deepl-connector/pom.xml:16]: https://axonivy.atlassian.net/browse/XIVY-10786

Comment (alexsuter) [reply deepl-connector/pom.xml:16]: The infos are already available.

File: deepl-connector-product/pom.xml (hunk 2 above)
Comment (alexsuter) [reply deepl-connector-product/pom.xml:47]: Or maybe we should have a variables editor which shows which variables can be overridden.

**Pull request merged**